core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -367,6 +367,9 @@ public void updateAddresses(final List<EquivalentAddressGroup> newAddressGroups)
     syncContext.execute(new Runnable() {
       @Override
       public void run() {
+        if (state.getState() == SHUTDOWN) {
```

Comment:
Does this do anything? If SHUTDOWN, the code seems fine; we'll update the addresses and nothing else.